packages/block-library/src/gallery/gallery.js

```diff
@@ -108,11 +111,43 @@ export const Gallery = ( props ) => {
 	);
 };
 
-function RichTextVisibilityHelper( { isHidden, ...richTextProps } ) {
-	return isHidden ? (
-		<VisuallyHidden as={ RichText } { ...richTextProps } />
-	) : (
-		<RichText { ...richTextProps } />
+function RichTextVisibilityHelper( {
+	isHidden,
+	captionFocused,
+	onFocusCaption,
+	className,
+	value,
+	placeholder,
+	tagName,
+	captionRef,
+	...richTextProps
+} ) {
+	if ( isHidden ) {
+		return <VisuallyHidden as={ RichText } { ...richTextProps } />;
+	}
+
+	if ( captionFocused ) {
+		return (
+			<RichText
+				ref={ captionRef }
+				value={ value }
+				placeholder={ placeholder }
+				className={ className }
+				tagName={ tagName }
+				isSelected={ captionFocused }
+				{ ...richTextProps }
+			/>
+		);
+	}
+
+	return (
+		<figcaption
+			role="presentation"
+			onClick={ onFocusCaption }
+			className={ className }
+		>
+			{ RichText.isEmpty( value ) ? placeholder : value }
+		</figcaption>
```

Comment:
Why is this needed? Why can't it be editable from the start?

Reply:
The rich text component was stealing focus as soon as the block was selected which was causing a few issues - https://github.com/WordPress/gutenberg/pull/31216